src/screens/crew6/relayScreen.cpp

```diff
@@ -162,6 +162,13 @@ RelayScreen::RelayScreen(GuiContainer* owner, bool allow_comms)
     });
     launch_probe_button->setSize(GuiElement::GuiSizeMax, 50)->setVisible(my_spaceship.hasComponent<ScanProbeLauncher>());
 
+    // Center on ship
+    center_button = new GuiButton(option_buttons, "CENTER_ON_SHIP", tr("Center On Ship"), [this](){
+        auto transform = my_spaceship.getComponent<sp::Transform>();
+        radar->setViewPosition(transform->getPosition());
```

Comment:
Ships that are internally docked do not have a transform. So you need to check if getComponent might return null before using it

Reply:
Thanks, I'll experiment with this!

Reply:
I've added null checks for everything and tested the internally docked case.